micro-rdk/src/common/ota.rs

```diff
@@ -103,20 +103,37 @@ struct EspAppDesc {
     reserv2: [u32; 20],
 }
 
+#[derive(Error, Debug)]
+pub(crate) enum ConfigError {
+    #[error("version {0} has length {1}, maximum allowed characters is {2}")]
+    InvalidVersionLen(String, usize, usize),
+    #[error("the given url {0}, is not a valid URI: {1}")]
+    InvalidUrl(String, String),
+    #[error(transparent)]
+    AttributeError(#[from] AttributeError),
+    #[error("config missing attribute `{0}`")]
+    MissingAttribute(String),
+    #[error("config missing value for `{0}`")]
```

Comment:
maybe `for field` if `{0}` is going to be the name of a present field (which I expect it would be).